PhysiCell/custom_modules/custom.cpp

```diff
@@ -276,8 +275,119 @@ void naive_bcell_phenotype( Cell* pCell, Phenotype& phenotype , double dt ) {
 }
+
+double alignment( Vector_Variable antigenSequence, Vector_Variable antibodySequence ) {
+
+    // strip and print input
+    std::vector<double> antigen_sequence {};
+    std::vector<double> antibody_sequence {};
+
+    for (double element : antigenSequence.value) {
+        if (element != pad) {
+            antigen_sequence.push_back(element);
+        }
+    }
+    for (double element : antibodySequence.value) {
+        if (element != pad) {
+            antibody_sequence.push_back(element);
+        }
+    }
+
+    printf("***antigen_sequence: ");
+    for (double element : antigen_sequence) {
+        printf("{%g}", element);
+    }
+    printf("***.\n");
+    printf("***antibody_sequence: ");
+    for (double element : antibody_sequence) {
+        printf("{%g}", element);
+    }
+    printf("***.\n");
+
+
+    // find smaller slide sequence and pad the longer one
+    std::vector<double> padded_sequence {};
+    std::vector<double> slide_sequence {};
+
+    int i_antigen = antigen_sequence.size();
+    int i_antibody = antibody_sequence.size();
+
+    if (i_antigen <= i_antibody) {
+        slide_sequence = antigen_sequence;
+        for (double element : antigen_sequence) {
+            padded_sequence.push_back(pad);
+        }
+        for (double element : antibody_sequence) {
+            padded_sequence.push_back(element);
+        }
+        for (double element : antigen_sequence) {
+            padded_sequence.push_back(pad);
+        }
+    }
+    else {
+        slide_sequence = antigen_sequence;
+        for (double element : antibody_sequence) {
+            padded_sequence.push_back(pad);
+        }
+        for (double element : antigen_sequence) {
+            padded_sequence.push_back(element);
+        }
+        for (double element : antibody_sequence) {
+            padded_sequence.push_back(pad);
+        }
+    }
+
+    // print padded sequence
+    printf("***padded_sequence: ");
+    for (double element : padded_sequence) {
+        printf("{%d}", (int) element);
+    }
+    printf("***.\n");
+    printf("***side_sequence: ");
+    for (double element : slide_sequence) {
+        printf("{%d}", (int) element);
+    }
+    printf("***.\n");
+
+
+    // get hamming distance.
+    double i_hammdist_max = 0.0;
+    int i_slide = slide_sequence.size();
+    int i_template = padded_sequence.size() - i_slide;
+
+    for (size_t i=0; i <= i_template; i++) {
+        double i_hammdist = 0.0;
+        printf("***sequence intersection: ");
+        for (size_t j=i; j < (i + i_slide); j++) {
+            if (padded_sequence[j] == slide_sequence[j-i]) {
+                i_hammdist = i_hammdist + 1.0;
+            }
+            // print slide sequences
+            printf("{%d}", (int) padded_sequence[j]);
+        }
+        printf("*** hamming distance: %g.\n", i_hammdist);
+        if (i_hammdist_max < i_hammdist) {
+            i_hammdist_max = i_hammdist;
+        }
+    }
```

Comment:
There is a way to write this loop so that there is no need for padded_sequence, but I'm glad you got it working nonetheless and won't make you change it.

Reply:
I am very interest to see these algorithms! Please provide you version. 
There is definitely something to learn for me.

Reply:
Some rough Python. I don't know if this is perfect, but you might find it interesting. 
```
best_hamm = -1
#i is where the slide window starts
for i in range(len(long_sequence)-len(slide), len(long_sequence)): 
  hamm = 0
  for j in range(len(slide)):
    if j < 0 or j > len(long_sequence):
        break #exit inner loop
    if long_sequence[i] == slide[j]:
      hamm += 1
  if hamm > best_hamm:
    best_hamm = hamm
```